src/fsharp/IlxGen.fs

```diff
@@ -4956,15 +5282,16 @@ and GetIlxClosureFreeVars cenv m (thisVars: ValRef list) eenvouter takenNames ex
     let eenvinner = eenvinner |> AddStorageForLocalVals g ilCloFreeVarStorage
 
     // Return a various results
-    (cloAttribs, cloFreeTyvars, cloWitnessInfos, cloFreeVars, ilCloTypeRef, ilCloAllFreeVars, eenvinner)
+    (cloFreeTyvars, cloWitnessInfos, cloFreeVars, ilCloTypeRef, ilCloAllFreeVars, eenvinner)
+
 
-and GetIlxClosureInfo cenv m isLocalTypeFunc canUseStaticField thisVars eenvouter expr =
+and GetIlxClosureInfo cenv m boxity isLocalTypeFunc canUseStaticField thisVars eenvouter expr =
     let g = cenv.g
     let returnTy =
       match expr with
       | Expr.Lambda (_, _, _, _, _, _, returnTy) | Expr.TyLambda (_, _, _, _, returnTy) -> returnTy
       | Expr.Obj (_, ty, _, _, _, _, _) -> ty
-      | _ -> failwith "GetIlxClosureInfo: not a lambda expression"
+      | _ -> tyOfExpr g expr //failwith "GetIlxClosureInfo: not a lambda expression"
```

Comment:
This intentional? Before we had a `failwith` and now it is commented out. If this is intentional then we should remove the commented `failwith`.

Reply:
It's intentional, because the expression could be a state machine expression.  Indeed the Expr.Obj line can be removed (see tyOfExpr definition for that case).